.github/workflows/go.yml

```
@@ -0,0 +1,28 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Build
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

```suggestion
on:
  push:
  pull_request:
    paths:
      - '**.go'
```
Only when Go files are modified?